src/main/java/Duke/Command/Parser.java

```diff
@@ -0,0 +1,269 @@
+/**
+ * @author Hang Zelin
+ *
+ * Parser will take in a full input Message and take out the operation type, task, time, index from the
+ * one line command input by users.
+ * It can also deal with the local saved data and return the parsed Message, which can be a task, time, done(or not).
+ * It can also parse the time users input into the LocalDateTime.
+ * Some invalid input Messages may cause throwing DukeException.
+ */
+
+package duke.command;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+
+import duke.excpetions.DukeException;
+import duke.task.TaskList;
+
+
+
+public class Parser {
+    private String message;
+
+    /**
+     * @param message Message users take in to be parsed.
+     */
+    public Parser(String message) {
+        this.message = message;
+    }
+
+
+    private static boolean isValidDate(int day, int month, int year, int hour, int minute) {
+        if (((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) && month == 2) {
```

Comment:
Great effort in doing your own validation checks! I just used LocalDateTime haha, anyhow might be good to break it apart for readability for others and also for yourself when you relook at the code. Additional boolean variables should go a long way :)

Reply:
Also good use of guard clauses 👍